src/test/java/org/jenkinsci/plugins/workflow/steps/scm/SCMStepTest.java

```diff
@@ -198,12 +198,8 @@ public void scmVars() throws Exception {
             WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
             p.setDefinition(new CpsFlowDefinition(
                     "node() {\n" +
-                            "  checkout(scm: [\n" +
-                            "    $class: 'GitSCM',\n" +
-                            "    branches: [[name: '*/master']],\n" +
-                            "    userRemoteConfigs: [[url: '" + sampleGitRepo.fileUrl() + "']]\n" +
-                            "  ])\n" +
-                            "}", true));
+                    "  git($/" + sampleGitRepo + "/$)\n" +
```

Comment:
Isn't this referencing sampleGitRepo before the repository has been initialized?

Or at least, possibly line 205 in gitChangelogSmokes is referencing `sampleGitRepo.fileUrl()` before the repo has been initialized?

Reply:
I thought that that might be a problem too, and tried moving it up last week, but it made no difference on my machine. From looking at the code, I think that `AbstractSampleDVCSRepoRule.toString` works fine even before `init` has been called since it just returns the absolute path to the folder that will be used for the repository, and that folder is initialized in an `@Before` method.

Reply:
I went ahead and moved `init` up in 9707099 for consistency and so that anyone who investigates the issue in the future won't have to think about this.